drivers/SmartThings/philips-hue/src/handlers/attribute_emitters.lua

```diff
@@ -125,30 +112,175 @@ local function _emit_light_events_inner(light_device, light_repr)
   end
 end
 
+function AttributeEmitters.connectivity_update(child_device, zigbee_status)
+  if zigbee_status.status == "connected" then
+    child_device.log.info_with({hub_logs=true}, "Light status event, marking device online")
+    child_device:online()
+    child_device:set_field(Fields.IS_ONLINE, true)
+  elseif zigbee_status.status == "connectivity_issue" then
+    child_device.log.info_with({hub_logs=true}, "Light status event, marking device offline")
+    child_device:set_field(Fields.IS_ONLINE, false)
+    child_device:offline()
+  end
+end
+
+function AttributeEmitters.emit_button_attribute_events(button_device, button_info)
+  if button_device == nil or (button_device and button_device.id == nil) then
+    log.warn("Tried to emit attribute events for a device that has been deleted")
+    return
+  end
+
+  if button_info.power_state then
+    log.debug(true, "emit power")
+    button_device:emit_event(
+      capabilities.battery.battery(
+        st_utils.clamp_value(button_info.power_state.battery_level, 0, 100)
+      )
+    )
+  end
+
+  local button_idx_map = button_device:get_field(Fields.BUTTON_INDEX_MAP)
+  if not button_idx_map then
```

Comment:
should there be a return in this if statement?

Reply:
I believe so, yes. Good catch.

Reply:
Fixed on https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1359